app/models/responsiblepeople/ResponsiblePerson.scala

```diff
@@ -140,18 +142,56 @@ case class ResponsiblePerson(personName: Option[PersonName] = None,
   def isComplete: Boolean = {
     Logger.debug(s"[ResponsiblePeople][isComplete] $this")
 
+    def hasToggleApprovalsDateVatAndAddress(pos: Positions, otherBusinessSP: Option[SoleProprietorOfAnotherBusiness]) = {
+      ApplicationConfig.phase2ChangesToggle &&
+      approvalFlags.hasAlreadyPassedFitAndProper.isDefined &
+      approvalFlags.hasAlreadyPaidApprovalCheck.isDefined &
+      pos.startDate.isDefined &
+      checkVatField(otherBusinessSP) &
+      validateAddressHistory
+    }
+
+    def hasToggleApprovalsDateVatAddressAndNoPreviousName(pName: PreviousName, pos: Positions, otherBusinessSP: Option[SoleProprietorOfAnotherBusiness]) = {
+      ApplicationConfig.phase2ChangesToggle &&
+      approvalFlags.hasAlreadyPassedFitAndProper.isDefined &
+      approvalFlags.hasAlreadyPaidApprovalCheck.isDefined &
+      pos.startDate.isDefined &
+      checkVatField(otherBusinessSP) &
+      validateAddressHistory &&
+      !pName.hasPreviousName.get
+    }
+
+    def hasNoToggle(pos: Positions, otherBusinessSP: Option[SoleProprietorOfAnotherBusiness]) = {
+      pos.startDate.isDefined &
+      checkVatField(otherBusinessSP) &
+      validateAddressHistory &
+      !ApplicationConfig.phase2ChangesToggle
+    }
+
+    def hasNoToggleAndNoPreviousName(pName: PreviousName, pos: Positions, otherBusinessSP: Option[SoleProprietorOfAnotherBusiness]) = {
+      pos.startDate.isDefined &
+      checkVatField(otherBusinessSP) &
+      validateAddressHistory &&
+      !pName.hasPreviousName.get &
+      !ApplicationConfig.phase2ChangesToggle
+    }
+
     this match {
-      case ResponsiblePerson(Some(_),Some(_),Some(_),Some(_),Some(_), _, _, Some(_),Some(_),Some(_), Some(pos),Some(_), _,Some(_),Some(_), Some(_), _, true, _, _, _, otherBusinessSP)
-        if pos.startDate.isDefined & checkVatField(otherBusinessSP) & validateAddressHistory & ApplicationConfig.phase2ChangesToggle => true
-      case ResponsiblePerson(Some(_),Some(pName),None,Some(_),Some(_), _, _, Some(_),Some(_),Some(_), Some(pos),Some(_), _,Some(_),Some(_), Some(_), _, true, _, _, _, otherBusinessSP)
-        if pos.startDate.isDefined & checkVatField(otherBusinessSP) & validateAddressHistory && !pName.hasPreviousName.get & ApplicationConfig.phase2ChangesToggle => true
+
+      case ResponsiblePerson(Some(_),Some(_),Some(_),Some(_),Some(_), _, _, Some(_),Some(_),Some(_), Some(pos),Some(_), _,Some(_),Some(_), _, _, true, _, _, _, otherBusinessSP)
```

Comment:
I don't think that this is as clear as it was previously - I can see that refactoring some of them together makes sense, but not all of them.

Reply:
ok